spec/observables/dom/ajax-spec.ts

```diff
@@ -1214,45 +1214,70 @@ class MockXMLHttpRequest {
     this.requestHeaders[key] = value;
   }
 
-  protected jsonResponseValue(response: any) {
-    try {
-      this.response = JSON.parse(response.responseText);
-    } catch (err) {
-      throw new Error('unable to JSON.parse: \n' + response.responseText);
-    }
-  }
-
-  protected defaultResponseValue() {
-    if (this.async === false) {
-      this.response = this.responseText;
+  respondWith(
+    response: {
+      status?: number;
+      responseText?: string | undefined;
+      responseType: XMLHttpRequestResponseType;
+      total?: number;
+      loaded?: number;
+    },
+    config?: { uploadProgressTimes?: number; downloadProgressTimes?: number }
+  ): void {
+    const { uploadProgressTimes = 0, downloadProgressTimes = 0 } = config ?? {};
+
+    // Fake our upload progress first, if requested by the test.
+    if (uploadProgressTimes) {
+      this.triggerUploadEvent('loadstart', { type: 'loadstart', total: uploadProgressTimes, loaded: 0 });
+      for (let i = 1; i <= uploadProgressTimes; i++) {
+        this.triggerUploadEvent('progress', { type: 'progress', total: uploadProgressTimes, loaded: i });
+      }
+      this.triggerUploadEvent('load', { type: 'load', total: uploadProgressTimes, loaded: uploadProgressTimes });
     }
-  }
 
-  respondWith(response: any, progressTimes?: number): void {
-    if (progressTimes) {
-      for (let i = 1; i <= progressTimes; ++i) {
-        this.triggerUploadEvent('progress', { type: 'ProgressEvent', total: progressTimes, loaded: i });
+    // Fake our download progress
+    if (downloadProgressTimes) {
+      this.triggerEvent('loadstart', { type: 'loadstart', total: downloadProgressTimes, loaded: 0 });
+      for (let i = 1; i <= downloadProgressTimes; i++) {
+        this.triggerEvent('progress', { type: 'progress', total: downloadProgressTimes, loaded: i });
       }
     }
+
+    // Set the readyState to 4.
     this.readyState = 4;
+
+    // Default to OK 200.
     this.status = response.status || 200;
     this.responseText = response.responseText;
-    const responseType = response.responseType !== undefined ? response.responseType : this.responseType;
-    if (!('response' in response)) {
```

Comment:
I could have left this I suppose, since I'm removing the property, but I think this `in` check was kind of confusing, so I opted to just let this `switch` process. And remove the `response` property again after.